clip_video_encode/clip_video_encode.py

```diff
@@ -120,36 +183,152 @@ def clip_video_encode(
         writer = FileWriter(dest)
     elif output_format == "webdataset":
         # TODO: maybe include params for this?
-        writer = WebDatasetWriter(dest, 9, "npy", maxcount=shard_sample_count, shard_id=starting_shard_id)
+        starting_shard_id = int(shards[0].split("/")[-1].split(".tar")[0])
+        writer = WebDatasetWriter(dest, 9, "npy", maxcount=1e6, shard_id=starting_shard_id)
 
     # Initialize model:
     model, _, preprocess = open_clip.create_model_and_transforms(oc_model_name, pretrained=pretrained, device=device)
     preprocess.transforms = [ToPILImage()] + preprocess.transforms[-3:]
-
     fm = FrameMapper(model, device)
-    fr = FrameReader(vids, meta_refs, take_every_nth, IMG_SIZE, workers=frame_workers, memory_size=frame_memory_size)
-    fr.start_reading()
 
-    frames, ind_dict = [], {}
-    block_size = 0
-    i = 0
-    for vid_frames, info in fr:
-        i += 1
-        frames.append(vid_frames)
-        ind_dict[info["reference"]] = (block_size, block_size + vid_frames.shape[0], info["dst_name"])
-        block_size += vid_frames.shape[0]
+    if input_format == "table":
+        fr = FrameReader(
+            vids,
+            meta_refs,
+            take_every_nth,
+            IMG_SIZE,
+            workers=frame_workers,
+            memory_size=frame_memory_size,
+        )
+        fr.start_reading()
+
+        frames, ind_dict = [], {}
+        block_size = 0
+        i = 0
+        for vid_frames, info in fr:
+            i += 1
+            frames.append(vid_frames)
+            ind_dict[info["reference"]] = (
+                block_size,
+                block_size + vid_frames.shape[0],
+                info["dst_name"],
+            )
+            block_size += vid_frames.shape[0]
+
+            if i % CHUNK_SIZE == 0:
+                encode_chunk(
+                    frames,
+                    ind_dict,
+                    writer,
+                    fm,
+                    preprocess,
+                    meta,
+                    ids,
+                    use_dst_name,
+                    device,
+                )
+                frames, ind_dict, block_size = [], {}, 0
 
-        if i % CHUNK_SIZE == 0:
-            encode_chunk(frames, ind_dict, writer, fm, preprocess, meta, ids, use_dst_name, device)
-            frames, ind_dict, block_size = [], {}, 0
+        if len(frames) > 0:  # TODO: make this cleaner
+            encode_chunk(
+                frames,
+                ind_dict,
+                writer,
+                fm,
+                preprocess,
+                meta,
+                ids,
+                use_dst_name,
+                device,
+            )
+    else:  # WebDataset shard logic
+        shard_times = []
+        for shard in shards:
+            times = {}
+            t = time.time()
+            try:
+                tempdir = tempfile.mkdtemp()
+                os.chmod(tempdir, 0o777)
```

Comment:
do you really need this? I've never had to do this? Also why .mkdtemp()? what's wrong with the normal ```with tempfile.TemporaryDirectory() as tmpdir:```?